trestle/core/markdown/markdown_node.py

```diff
@@ -74,6 +74,11 @@ def get_all_headers_for_level(self, level: int) -> Iterable[str]:
             filter(lambda header: self._get_header_level_if_valid(header) == level, self.content.subnodes_keys)
         ).__iter__()
 
+    def get_all_headers_for_level_and_starting_key(self, level: int, key: str) -> List[str]:
+        """Return all headers at the specified header level and starting with the key."""
+        level_headers = self.get_all_headers_for_level(level)
+        return [header for header in level_headers if header.startswith('## ' + key + ' ')]
```

Comment:
This line will not do what function should do. 

If I do `get_all_headers_for_level_and_starting_key(3, 'Some key')` it will not match anything. Because of the `header.startswith('## ' + key + ' ')`, it will never match `### Some key` only `## Some key`. 
Same goes for `get_all_headers_for_level_and_starting_key(1, 'Some key')`

Another important point is that sometimes there is no space after `###` or after the key.

I believe you can simply check for the substring existence here. I.e. `[header for header in level_headers if key in header]` 

If more stricter version is needed, please use regex.

Reply:
Thanks - I have removed the function since it is very specialized and now make a strict check for the exact header prefix including the one space - since it is generated by trestle.  Previously it was finding any level 2 header with the word Implementation in it - which threw things off the rails since it wasn't strict enough.